tools/StaticAnalysis/BreakingChangeAnalyzer/TypeMetadataHelper.cs

```diff
@@ -0,0 +1,481 @@
+                                oldTypeMetadata.Name, parameter.Name,
+                                oldTypeMetadata.GenericTypeArguments.Count, newTypeMetadata.GenericTypeArguments.Count),
+                            remediation: string.Format(Properties.Resources.DifferentParameterGenericTypeArgumentSizeRemediation,
+                                oldTypeMetadata.Name, oldTypeMetadata.GenericTypeArguments.Count));
+                    }
+                }
+                // If the generic type has changed, log an issue
+                else
+                {
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: cmdlet,
+                        severity: 0,
+                        problemId: ProblemIds.BreakingChangeProblemId.ChangedParameterGenericType,
+                        description: string.Format(Properties.Resources.ChangedParameterGenericTypeDescription,
+                            parameter.Name, oldTypeMetadata.Name, newTypeMetadata.Name),
+                        remediation: string.Format(Properties.Resources.ChangedParameterGenericTypeRemediation,
+                            parameter.Name, oldTypeMetadata.Name));
+                }
+
+                return;
+            }
+
+            // If the types are different, log an issue
+            if (!oldTypeMetadata.Name.Equals(newTypeMetadata.Name, StringComparison.OrdinalIgnoreCase))
+            {
+                issueLogger.LogBreakingChangeIssue(
+                    cmdlet: cmdlet,
+                    severity: 0,
+                    problemId: ProblemIds.BreakingChangeProblemId.ChangedParameterType,
+                    description: string.Format(Properties.Resources.ChangedParameterTypeDescription,
+                        cmdlet.Name, oldTypeMetadata.Name, parameter.Name),
+                    remediation: string.Format(Properties.Resources.ChangedParameterTypeRemediation,
+                        parameter.Name, oldTypeMetadata.Name));
+            }
+            else
+            {
+                CompareTypeMetadata(cmdlet, oldTypeMetadata, newTypeMetadata, issueLogger);
+            }            
+        }
+
+        /// <summary>
+        /// Checks if the type of the output is an array or a generic, and makes sure there are no breaking changes.
+        /// If the type is not an array or a generic, it proceeds with the normal type checking with CompareTypeMetadata.
+        /// </summary>
+        /// <param name="cmdlet">The cmdlet whose output metadata is being checked for breaking changes.</param>
+        /// <param name="oldTypeMetadata">The type metadata from the old (serialized) assembly.</param>
+        /// <param name="newTypeMetadata">The type metadata from the new assembly.</param>
+        /// <param name="issueLogger">ReportLogger that will keep track of issues found.</param>
+        public void CheckOutputType(
```

Comment:
same comment - this should be broken down into the individual types of comparison, and we should look for opportunities to reuse code so there is less duplication